packages/mds-ingest-service/repository/index.ts

```diff
@@ -473,21 +473,23 @@ class IngestReadWriteRepository extends ReadWriteRepository {
   public getLatestTelemetryForDevices = async (device_ids: UUID[]): Promise<TelemetryDomainModel[]> => {
     try {
       const connection = await this.connect('ro')
+      /* sets enable_bitmapscan = off for this connection the duration of the session */
+      await connection.getRepository(TelemetryEntity).query(`select set_config('enable_bitmapscan','off', false)`)
+
       const entities = await connection
         .createQueryBuilder(TelemetryEntity, 'telemetry')
-        .innerJoin(
-          subquery =>
-            subquery
-              .select('id, RANK() OVER (PARTITION BY device_id ORDER BY timestamp DESC) AS rownum')
-              .where('device_id = ANY(:device_ids)', { device_ids })
-              .from(TelemetryEntity, 't'),
-          'last_device_telemetry',
-          'last_device_telemetry.id = telemetry.id AND last_device_telemetry.rownum = 1'
-        )
+        .distinctOn(['device_id'])
```

Comment:
Just looked up what this did, pretty cool.

Reply:
thanks!